R/isoread_rds.R

```diff
@@ -1,17 +1,15 @@
 # read cached isoreader files
 # @param ds the data structure to fill
-iso_read_rda <- function(ds) {
-  
+iso_read_rds <- function(ds) {
+
   # safety checks
   if(!iso_is_file(ds)) stop("data structure must be an iso_file", call. = FALSE)
-  
-  # load rda file
-  if (exists("iso_files", inherits = FALSE)) rm("iso_files")
-  if (exists("isofiles", inherits = FALSE)) rm("isofiles") # backwards compatibility
-  load(ds$file_info$file_path) 
-  
+
+  # load rds file
+  iso_files  <- readRDS(ds$file_info$file_path)
+
   # make sure object in file was loaded properly
-  if (exists("isofiles", inherits = FALSE)) { 
+  if (exists("isofiles", inherits = FALSE)) {
```

Comment:
this line is no longer necessary since `readRDS` assigns directly to `iso_files`, however the next couple of safety lines will be good to have for the old `iso_read_rda` function but not necessary for the `iso_read_rds` since writeRDS was only used starting at this version and there is no need for back-wards compatibility